docs/ref/urls.txt

```diff
@@ -5,7 +5,116 @@
 .. module:: django.urls.conf
     :synopsis: Functions for use in URLconfs.
 
-.. currentmodule:: django.conf.urls
+.. currentmodule:: django.urls
+
+``path()``
+==========
+
+.. function:: path(route, view, kwargs=None, name=None)
+
+.. versionadded:: 2.0
+
+Returns an element for inclusion in ``urlpatterns``. For example::
+
+    from django.urls import include, path
+
+    urlpatterns = [
+        path('index/', index_view, name='main-view'),
+        path('personal/<username>/', views.personal_page),
+        path('articles/<slug:title>/', views.article),
+        path('articles/<slug:title>/<int:section>/', views.section),
+        path('weblog/', include('blog.urls')),
+        ...
+    ]
+
+The ``route`` parameter should be a string or
+:func:`~django.utils.translation.gettext_lazy()` (see
+:ref:`translating-urlpatterns`) that contains a URL path pattern. A pattern is
+a prefix of the path to be matched, but it may also pick parts of the path to
+be captured and passed as parameters to the view when a match is made. Such
+parts, or fields, are specified by a name in angled brackets (like
+``<username>`` above), with an optional converter specification (like
+``<int:section>``). Where converters are specified, they also imply a format
+for the field -- e.g. ``<int:section>`` will only match a string of decimal
+digits. See :ref:`How Django processes a
+request<how-django-processes-a-request>` for further discussion.
+
+The ``view`` parameter is a view function or the result of
+:meth:`~django.views.generic.base.View.as_view` for class-based views. It can
+also be an :func:`django.urls.include`.
+
+The ``kwargs`` parameter allows you to pass additional arguments to the view
+function or method. See :ref:`views-extra-options` for an example.
+
+See :ref:`Naming URL patterns <naming-url-patterns>` for why the ``name``
+parameter is useful.
+
+``re_path()``
+=============
+
+.. function:: re_path(route, view, kwargs=None, name=None)
+
+.. versionadded:: 2.0
+
+Returns an element for inclusion in ``urlpatterns``. For example::
+
+    from django.urls import include, re_path
+
+    urlpatterns = [
+        re_path(r'^index/$', index_view, name='main-view'),
+        re_path(r'^personal/(?P<username>\w+)/$', views.personal_page),
+        re_path(r'^weblog/', include('blog.urls')),
+        ...
+    ]
+
+The ``route`` parameter should be a string or
+:func:`~django.utils.translation.gettext_lazy()` (see
+:ref:`translating-urlpatterns`) that contains a regular expression compatible
+with Python's :py:mod:`re` module. Strings typically use raw string syntax
+(``r''``) so that they can contain sequences like ``\d`` without the need to
+escape the backslash with another backslash. When a match is made, captured
+groups from the regular expression are passed to the view -- as named arguments
+if the groups are named, and as positional arguments otherwise; the values are
+passed as strings, with no type conversion.
+
```

Comment:
(Might) need a note that if named capture groups are defined, the non-named capture groups are not used.